plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/jmx/OpenshiftConnectionWrapper.java

```diff
@@ -0,0 +1,150 @@
+import javax.management.remote.JMXServiceURL;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.wst.server.core.IServer;
+import org.jboss.tools.jmx.core.IConnectionProvider;
+import org.jboss.tools.jmx.core.IConnectionWrapper;
+import org.jboss.tools.jmx.core.IJMXRunnable;
+import org.jboss.tools.jmx.core.JMXActivator;
+import org.jboss.tools.jmx.core.JMXCoreMessages;
+import org.jboss.tools.jmx.core.JMXException;
+import org.jboss.tools.jmx.core.tree.NodeUtils;
+import org.jboss.tools.jmx.core.tree.Root;
+import org.jboss.tools.openshift.core.server.OpenShiftServerUtils;
+import org.jboss.tools.openshift.internal.core.portforwarding.PortForwardingUtils;
+import org.jboss.tools.openshift.internal.core.server.debug.DebuggingContext;
+import org.jboss.tools.openshift.internal.core.server.debug.OpenShiftDebugUtils;
+
+import com.openshift.restclient.model.IDeploymentConfig;
+import com.openshift.restclient.model.IPod;
+
+public class OpenshiftConnectionWrapper implements IConnectionWrapper {
+	private JMXConnector connector;
+	private MBeanServerConnection connection;
+	private Root root;
+	private boolean isLoading;
+	private boolean isConnected;
+	private IServer server;
+	private OpenshiftConnectionProvider provider;
+	private JMXConnectorProvider factory;
+
+	public OpenshiftConnectionWrapper(OpenshiftConnectionProvider provider, IServer server,
+			JMXConnectorProvider factory) throws MalformedURLException {
+		this.provider = provider;
+		this.server = server;
+		this.factory = factory;
+		this.isConnected = false;
+		this.isLoading = false;
+	}
+
+	public IConnectionProvider getProvider() {
+		return provider;
+	}
+
+	public String getName() {
+		return "Connection for " + server.getId();
+	}
+
+	public boolean canControl() {
+		return true;
+	}
+
+	public synchronized void connect() throws IOException {
+		try {
+			IDeploymentConfig dc = OpenShiftServerUtils.getDeploymentConfig(server);
+			DebuggingContext debugContext = OpenShiftDebugUtils.get().getDebuggingContext(dc);
+			HashMap<String, Object> env = new HashMap<>();
+			env.put(JMXConnector.CREDENTIALS,
+					new String[] { debugContext.getAdminUsername(), debugContext.getAdminPassword() });
+			IPod firstPod = OpenShiftDebugUtils.get().getFirstPod(dc);
+			if (firstPod != null) {
+				int port = getLocalAdminPort(firstPod);
+				String url = "service:jmx:remoting-jmx://localhost:" + port;
+				// try to connect
+				connector = factory.newJMXConnector(new JMXServiceURL(url), env);
+				connector.connect();
+				connection = connector.getMBeanServerConnection();
+				isConnected = true;
+				provider.fireChanged(this);
+			}
+		} catch (CoreException e) {
+			throw new IOException(e);
+		}
+	}
+
+	public synchronized void disconnect() throws IOException {
+		// close
+		root = null;
+		isConnected = false;
```

Comment:
shouldnt we set isConnected to false only once we successfully closed the connection?